nixos/modules/services/networking/tailscale-cert.nix

```diff
@@ -0,0 +1,143 @@
+{ config, options, lib, pkgs, ... }:
+
+let
+  cfg = config.services.tailscale-cert;
+  opt = options.services.tailscale-cert;
+in
+{
+  options.services.tailscale-cert = {
+    enable = lib.mkEnableOption "the Tailscale cert service";
+
+    package = lib.mkPackageOption pkgs "tailscale" { };
+
+    domain = lib.mkOption {
+      type = lib.types.str;
+      description = "Tailnet domain for which to request the certificate";
+      example = "monitoring.tailnet0123.ts.net";
+    };
+
+    dataDir = lib.mkOption {
+      type = lib.types.str;
+      default = "tailscale-cert";
+      description = ''
+        Directory name under /var/lib/ where to store certificates and keys.
+      '';
+    };
+
+    certFile = lib.mkOption {
+      type = lib.types.str;
+      default = "${cfg.domain}.crt";
+      defaultText = lib.literalExpression "config.${opt.domain}.crt";
```

Comment:
Is the "config." prefix here (and in keyFile) intentional? If so, what purpose does it serve?

(type could be lib.types.path)

Reply:
I hadn't noticed that the value of `defaultText` wasn't correct. Now it looks like this, which I think is correct:

```
services.tailscale-cert.certFile
           File name for cert under <option>services.tailscale-cert.dataDir</option>

           Type: string

           Default: "${config.services.tailscale-cert.domain}.crt"
```

Does this make sense?

Reply:
> (type could be lib.types.path)

I don't think so, because this is just the file name? The destination is hard-coded to be under /var/lib, with the directory name being defined by `dataDir`.